src/components/common/ChipInput/ChipInput.tsx

```diff
@@ -0,0 +1,57 @@
+'use client';
+import React, { CSSProperties, useCallback } from 'react';
+import * as styles from './ChipInput.css';
+import Chip from './Chip';
+
+export type ChipOption = {
+  value: string;
+  label: string;
+};
+interface ChipInputProps {
+  name: string;
+  value: string;
+  options: ChipOption[] | string[];
+  onInput: (name: string, value: string) => void;
+  style?: CSSProperties;
+}
+
+const ChipInput: React.FC<ChipInputProps> = ({
+  options,
+  onInput,
+  name,
+  value,
+  style
+}) => {
+  const getOptionValue = useCallback((option: (typeof options)[0]) => {
+    return typeof option === 'string' ? option : option.value;
+  }, []);
+
+  const getOptionLabel = useCallback((option: (typeof options)[0]) => {
+    return typeof option === 'string' ? option : option.label;
+  }, []);
+
+  const handleClickChip = useCallback(
+    (option: string | ChipOption) => {
+      onInput(name, getOptionValue(option));
+    },
+    [getOptionValue, name, onInput]
+  );
```

Comment:
```suggestion
 
```suggestion
       const handleClickChip = useCallback(
    (option: string | ChipOption) => () => {
      onInput(name, getOptionValue(option));
    },
    [getOptionValue, name, onInput]
  );
```
이런식으로 수정하면 < Chip/> 컴포넌트에서  children을 인자로 넣어주지 않아도 될거같습니다!

Reply:
너무 좋네요 감사합니다!!